app/src/main/java/com/android/streetworkapp/ui/navigation/TopAppBarManager.kt

```diff
@@ -3,16 +3,47 @@ package com.android.streetworkapp.ui.navigation
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.automirrored.filled.ArrowBack
 import androidx.compose.ui.graphics.vector.ImageVector
+import com.android.sample.R
 
 class TopAppBarManager(
     private var title: String = "",
     private var hasNavigationIcon: Boolean = false,
-    private var navigationIcon: ImageVector? = null
+    private var navigationIcon: ImageVector? = null,
+    private var actions: List<TopAppBarAction> = emptyList(),
+    private var actionCallbacks: MutableMap<TopAppBarAction, () -> Unit> = mutableMapOf()
 ) {
 
   companion object {
     val DEFAULT_TOP_APP_BAR_NAVIGATION_ICON = Icons.AutoMirrored.Filled.ArrowBack
   }
+
+  enum class TopAppBarAction(val icon: Int, val contentDescription: String, val testTag: String) {
+    SETTINGS(
+        icon = R.drawable.settings, contentDescription = "Settings", testTag = "settings_button");
+
+    // Now `onClick` is a parameter for `TopAppBarAction`
+    fun onClick(callback: () -> Unit) {
+      callback()
```

Comment:
Does this function do anything ?
I removed it on my side and nothing changed

Reply:
It is an older version of the callbacks for actions, because I am still implementing (see "**Ongoing PR - do not review**" section). It actually does nothing currently because callbacks are handled through a mapping `actionCallbacks: MutableMap<TopAppBarAction, () -> Unit>`.